lib/init.c

```diff
@@ -187,7 +191,25 @@ pci_alloc(void)
   int i;
 
   memset(a, 0, sizeof(*a));
+#ifdef PCI_OS_WINDOWS
+  if ((PCI_PATH_IDS_DIR)[0])
+    pci_set_name_list_path(a, PCI_PATH_IDS_DIR "\\" PCI_IDS, 0);
+  else
+    {
+      char *path = malloc(MAX_PATH+1);
+      DWORD len = GetModuleFileNameA(NULL, path, MAX_PATH+1);
```

Comment:
Crashes when `malloc` returns `NULL`. Use `pci_malloc` instead?

Reply:
`pci_malloc` cannot be used at this place as `a->error` (used by `pci_malloc`) is not initialized yet.

Reply:
fixed

Reply:
I don't want to have many copies of an out of memory message. But is there any reason for `malloc`ing the buffer instead of having it as a local variable?

Reply:
`pci_set_name_list_path()` takes pointer of buffer for later usage, so `path` needs to be either global/static variable or `malloc`ed buffer. `pci_set_name_list_path()` API is designed for `malloc`ed buffers (third argument specify if libpci will call `free()`), so I chose to use `malloc()`. Do you prefer something else (static or global variable)?

Reply:
Ah yes...

I changed `pci_malloc()`, so that it can be called even if `pci_access` is not fully initialized yet. You can now use it here.

Reply:
done!